### PR TITLE
WIP: AsyncIO integration

### PR DESCRIPTION
**Pull request by lidizheng**
Let's see if tests are happy. Somehow I cannot run tests against all Python versions locally.

---

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-api-core/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by lidizheng** [comment]
@busunkim96 Can you help me trigger the test one more time? I think I resolved the test coverage and linter complains.

File: noxfile.py [hunk 1]
```diff
@@ -32,19 +45,49 @@ def default(session):
...
-        "--cov=tests.unit",
-        "--cov-append",
-        "--cov-config=.coveragerc",
-        "--cov-report=",
-        "--cov-fail-under=0",
-        os.path.join("tests", "unit"),
-        *session.posargs
-    )
+    # Inject AsyncIO content, if version >= 3.6.
+    if _greater_or_equal_than_36(session.python):
```
Comment (busunkim96) [inline noxfile.py:49]: Does a string comparison miss some cases?

```py
if session.python >= '3.6'
```

You can also use [packaging.version.parse](https://packaging.pypa.io/en/latest/version/#packaging.version.parse).

Comment (lidizheng) [reply noxfile.py:49]: I was using `packaging.version`, but Python in CI doesn't have the `packaging` package.
```
ModuleNotFoundError: No module named 'packaging'
```
https://source.cloud.google.com/results/invocations/d91e385b-d996-405c-9c94-4ba26e26abad/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-api-core%2Fpresubmit%2Fpresubmit/log

I am afraid of two corner case: 1) Python goes to 4.0; 2) Python goes to 3.10+. In second corner case, the simple string comparison fails.

**Comment by busunkim96** [comment]
Closing as this is made obsolete by #26, #28 , #29

**Pull request closed without merging**